domain/src/main/java/no/difi/meldingsutveksling/receipt/ConversationService.java

```diff
@@ -1,161 +1,179 @@
+                String body = format("%s forsendelse med conversationId %s (messageId %s) %shar registrert status '%s'. Se statusgrensesnitt for detaljer.",
+                        direction, conversation.getConversationId(), conversation.getMessageId(), messageRef, FEIL.toString());
+                mailSender.send(title, body);
+            } catch (Exception e) {
+                log.error(format("Error sending status mail for messageId %s", conversation.getMessageId()), e);
             }
-            return repo.save(conversation);
         }
+        if (asList(LEST, FEIL, LEVETID_UTLOPT, INNKOMMENDE_LEVERT).contains(ReceiptStatus.valueOf(status.getStatus()))) {
+            conversation.setFinished(true)
+                    .setPollable(false);
+        }
+
+        log.debug(String.format("Added status '%s' to conversation[id=%s]", status.getStatus(),
+                conversation.getMessageId()),
+                MessageStatusMarker.from(status));
+
+        repo.save(conversation);
+
+        webhookPublisher.publish(conversation, status);
+
         return conversation;
     }
 
-    public Conversation markFinished(Conversation conversation) {
-        conversation.setFinished(true);
-        conversation.setPollable(false);
+    private boolean isPollable(MessageStatus status) {
+        Conversation conversation = status.getConversation();
+        return conversation.getDirection() == ConversationDirection.OUTGOING &&
+                ReceiptStatus.SENDT.toString().equals(status.getStatus()) &&
+                POLLABLES.contains(conversation.getServiceIdentifier());
+    }
+
+    @Transactional
+    public Conversation save(Conversation conversation) {
         return repo.save(conversation);
     }
 
-    public Conversation registerConversation(EDUCore message) {
-        Optional<Conversation> find = repo.findByConversationId(message.getId()).stream().findFirst();
-        if (find.isPresent()) {
-            log.warn(format(CONVERSATION_EXISTS, message.getId()));
-            return find.get();
-        }
+    @Transactional
+    public Conversation registerConversation(MessageInformable message) {
+        return findConversation(message.getMessageId())
+                .orElseGet(() -> createConversation(message));
+    }
 
-        MessageStatus ms = MessageStatus.of(GenericReceiptStatus.OPPRETTET);
-        if (message.getMessageType() == EDUCore.MessageType.APPRECEIPT) {
-            ms.setDescription("AppReceipt");
-        }
-        Conversation conversation = Conversation.of(message, ms);
+    @Transactional
+    public Conversation registerConversation(StandardBusinessDocument sbd, ServiceIdentifier si, ConversationDirection conversationDirection) {
+        OffsetDateTime ttl = sbd.getExpectedResponseDateTime().orElse(OffsetDateTime.now(clock).plusHours(props.getNextmove().getDefaultTtlHours()));
+        return registerConversation(new MessageInformable() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 30 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1188)